ansible_base/authentication/social_auth.py

```diff
@@ -189,12 +192,53 @@ def validate(self, serializer, data):
         return data
 
 
+def extract_id_token_claims(backend, response):
+    if 'id_token' not in response:
+        return {}
+
+    try:
+        id_token = response['id_token']
+        client_id, _ = backend.get_key_and_secret()
+
+        key = backend.find_valid_key(id_token)
```

Comment:
Will this always be defined for all authenticator plugins calling this? No base class that I can locate in my dependencies seems to have `find_valid_key` defined.

Reply:
It's implemented in `social_core.backends.open_id_connect::OpenIdConnectAuth` https://github.com/python-social-auth/social-core/blob/master/social_core/backends/open_id_connect.py#L175.

This is essentially only the case for the `oidc` authenticator plugin.

I think we probably actually want to move this into the authenticator plugin instead of here.

Let me have a go at that